docs/sdk/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,618 @@
+//!
+//! <details>
+//!   <summary><b>Solution: Saturating or Checked</b></summary>
+//!     For Bob's balance problems, using a `saturating_add` or `checked_add` could've mitigated
+//! this issue.  They simply would've reached the upper, or lower bounds, of the particular type for
+//! an on-chain balance.  In other words: Bob's balance would've stayed at the maximum of the
+//! Balance type. </details>
+//!
+//! #### Alice's 'Underflowed' Balance
+//!
+//! Alice's balance has reached `0` after a transfer to Bob. Suddenly, she has been slashed on
+//! `EduChain`, causing her balance to reach near the limit of `u32::MAX` - a very large amount - as
+//! _wrapped operations_ can go both ways. Alice can now successfully vote using her new,
+//! overpowered token balance, destroying the integrity of the chain.
+//!
+//! <details>
+//!   <summary><b>Solution: Saturating</b></summary>
+//!   For Alice's balance problem, using `saturated_sub` could've mitigated this issue.  As debt or
+//!   having a negative balance is not a concept within blockchains, a saturating calculation
+//! would've simply limited her balance to the lower bound of u32.
+//!
+//!   In other words: Alice's balance would've stayed at "0", even after being slashed.
+//!
+//!   This is also an example that while one system may work in isolation, shared interfaces, such
+//!   as the notion of balances, are often shared across multiple pallets - meaning these small
+//!   changes can make a big difference in outcome. </details>
+//!
+//! #### Proposals' ID Overwrite
+//!
+//! The type for counting the number of proposals on-chain is represented by a `u8` number, called
+//! `proposals_count`. Every time a new proposal is added to the system, this number increases. With
+//! the proposal pallet being high in usage, it has reached `u8::MAX`'s limit of `255`, causing
+//! `proposals_count` to go to `0`. Unfortunately, this resulted in new proposals overwriting old
+//! ones, effectively erasing any notion of past proposals!
+//!
+//! <details>
+//!  <summary><b>Solution: Checked</b></summary>
+//! For the proposal IDs, proper handling via `checked` math would've been much more suitable,
+//! Saturating could've been used - but it also would've 'failed' silently. Using `checked_add` to
+//! ensure that the next proposal ID would've been valid would've been a viable way to let the user
+//! know the state of their proposal:
+//!
+//! ```ignore
+//! let next_proposal_id = current_count.checked_add(1).ok_or_else(|| Error::TooManyProposals)?;
+//! ```
+//!
+//! </details>
+//!
+//! From the above, we can clearly see the problematic nature of seemingly simple operations in the
+//! runtime. Of course, it may be that using unchecked math is perfectly fine under some scenarios -
+//! such as certain balance being never realistically attainable, or a number type being so large
+//! that it could never realistically overflow unless one sent thousands of transactions to the
+//! network.
```

Comment:
Neither of the presented scenarios warrant usage of unchecked math. If there are situtations that limit someone from using checked math, they should be highlighted here. Else, we don't say anything.

```suggestion
//! runtime. 
```